lib/ProductOpener/Tags.pm

```diff
@@ -4393,14 +4463,17 @@ sub generate_regexps_matching_taxonomy_entries ($taxonomy, $return_type, $option
 
 	foreach my $tagid (get_all_taxonomy_entries($taxonomy)) {
 
-		foreach my $language (keys %{$translations_to{$taxonomy}{$tagid}}) {
+		foreach my $language (sort keys %{$translations_to{$taxonomy}{$tagid}}) {
 
 			defined $synonyms_regexps{$language} or $synonyms_regexps{$language} = [];
 
 			# the synonyms below also contain the main translation as the first entry
 
 			foreach my $synonym (get_taxonomy_tag_synonyms($language, $taxonomy, $tagid)) {
 
+				# Escape some characters
+				$synonym =~ s/([\(\)\/\?\*\[\]\{\}])/\\$1/g;
```

Comment:
we have a similar one in another module indeed:

`lib/ProductOpener/ImportConvert.pm:					$brand =~ s/(\*|\+|\?|\(|\)|\[|\]|\{|\}|\$|\^|\\)/\\$1/g;
`

Instead of a global variable, I think here it's best to have a function escape_special_regexp_chars()

Reply:
I'll add a regexp_escape() in Text.pm